parser/types.go

```diff
@@ -119,29 +119,30 @@ func incompletePipeExpr(call interface{}, text []byte, pos position) (*ast.PipeE
 func memberexprs(head, tail interface{}, text []byte, pos position) (ast.Expression, error) {
 	res := head.(ast.Expression)
 	for _, prop := range toIfaceSlice(tail) {
-		res = &ast.MemberExpression{
-			Object:   res,
-			Property: prop.(ast.Expression),
-			BaseNode: base(text, pos),
+		switch expr := prop.(type) {
+		case *ast.MemberExpression:
+			expr.BaseNode = base(text, pos)
+			expr.Object = res
+			res = expr
+		case *ast.IndexExpression:
```

Comment:
I am not sure why `memberexprs` changes? How does an IndexExpression get passed to this method?

Reply:
The grammar defines a member expression as an identifier followed by a sequence of `MemberExpressionProperty`s. These properties can either be of the form `.identifier` or `[expression]`.